Scripts/Editor/Core/CollectionItemIndirectReferencePropertyDrawer.cs

```diff
@@ -50,50 +65,68 @@ public override void OnGUI(Rect position, SerializedProperty property, GUIConten
             EditorGUI.PropertyField(position, property, label, true);
         }
 
-        private void DrawItemDrawer(Rect position, GUIContent label, ScriptableObjectCollectionItem collectionItem
+        private void DrawItemDrawer(Rect position, GUIContent label, ScriptableObject collectionItem
         )
         {
-            collectionItemPropertyDrawer.DrawCollectionItemDrawer(ref position, collectionItem, label, item =>
+            socItemPropertyDrawer.DrawCollectionItemDrawer(ref position, collectionItem, label, item =>
             {
                 SetSerializedPropertyGUIDs(item);
                 drawingProperty.serializedObject.ApplyModifiedProperties();
             });
         }
 
-        private void SetSerializedPropertyGUIDs(ScriptableObjectCollectionItem item)
+        private void SetSerializedPropertyGUIDs(ScriptableObject item)
         {
             if (item == null)
             {
-                itemGUIDSerializedProperty.stringValue = string.Empty;
-                collectionGUIDSerializedProperty.stringValue = string.Empty;
+                itemGUIDValueASerializedProperty.longValue = 0;
+                itemGUIDValueBSerializedProperty.longValue = 0;
+                collectionGUIDValueASerializedProperty.longValue = 0;
+                collectionGUIDValueBSerializedProperty.longValue = 0;
+                
             }
             else
             {
-                itemGUIDSerializedProperty.stringValue = item.GUID;
-                collectionGUIDSerializedProperty.stringValue = item.Collection.GUID;
+                if (item is ISOCItem socItem)
+                {
+                    (long, long) itemGUIDValues = socItem.GUID.GetValue();
+                    itemGUIDValueASerializedProperty.longValue = itemGUIDValues.Item1;
+                    itemGUIDValueBSerializedProperty.longValue = itemGUIDValues.Item2;
+
+                    (long, long) collectionGUIDValues = socItem.Collection.GUID.GetValue();
+                    collectionGUIDValueASerializedProperty.longValue = collectionGUIDValues.Item1;
+                    collectionGUIDValueBSerializedProperty.longValue = collectionGUIDValues.Item2;
+                }
             }
         }
 
-        private static ScriptableObjectCollectionItem GetCollectionItem(string itemGUID, string collectionGUID)
+        private bool TryGetCollectionItem(out ScriptableObject item)
         {
-            if (string.IsNullOrEmpty(itemGUID)) 
-                return null;
+            item = null;
+
+            if (itemGUIDValueASerializedProperty.longValue == 0 ||
+                itemGUIDValueBSerializedProperty.longValue == 0 ||
+                collectionGUIDValueASerializedProperty.longValue == 0 ||
+                collectionGUIDValueBSerializedProperty.longValue == 0)
+                return false;
```

Comment:
I personally find it more readable to give the statement under a multi-line condition braces even if it's only one line.

At Paladin that's mandatory, don't know about the code style at Moon. Worth considering but up to your personal style.

Reply:
Ow yeah, this looks awful without braces, I will fix it!